// @crossmint/client-sdk-react-ui/lib/cjs/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mintingContractTypes = exports.customHeaders = exports.baseUrls = exports.clientNames = void 0;
var clientNames;
(function (clientNames) {
    clientNames["reactUi"] = "client-sdk-react-ui";
})(clientNames = exports.clientNames || (exports.clientNames = {}));
var baseUrls;
(function (baseUrls) {
    baseUrls["prod"] = "https://www.crossmint.io";
    baseUrls["dev"] = "http://localhost:3001";
})(baseUrls = exports.baseUrls || (exports.baseUrls = {}));
var customHeaders;
(function (customHeaders) {
    customHeaders["clientVersion"] = "X-Client-Version";
    customHeaders["clientName"] = "X-Client-Name";
})(customHeaders = exports.customHeaders || (exports.customHeaders = {}));
var mintingContractTypes;
(function (mintingContractTypes) {
    mintingContractTypes["CANDY_MACHINE"] = "candy-machine";
    mintingContractTypes["SOLANA_AUCTION"] = "solana-auction";
    mintingContractTypes["ERC_721"] = "erc-721";
})(mintingContractTypes = exports.mintingContractTypes || (exports.mintingContractTypes = {}));
